examples/generative/Density estimation using RealNVP.py

```diff
@@ -0,0 +1,251 @@
+and whose density is simple to estimate - to a more complex one learned from the data;
+these kind of generative models is also knowm as "normalizing flows".
+
+In order to obtain the aforementioned result the model is trained by means of the maximum
+likelihood principle, resorting to the "change of variable formula".
+
+An affine coupling layer function is built. It is created such that its inverse and the determinant
+of the jacobian is easy to obtain (more details in the referenced paper).
+
+requirements:
+
+* Tensorflow 2.3.0
+* Tensorflow probability 0.11.0
+
+References:
+[Density estimation using Real NVP](https://arxiv.org/pdf/1605.08803.pdf)
+"""
+
+"""
+## Setup
+
+"""
+
+# libraries to import
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+from tensorflow.keras import regularizers
+from sklearn.preprocessing import StandardScaler
+import numpy as np
+import matplotlib.pyplot as plt
+import tensorflow_probability as tfp
+
+"""
+## Load the data
+"""
+
+# Generating the examples
+
+url = "http://cs.joensuu.fi/sipu/datasets/s1.txt"
+path = tf.keras.utils.get_file("dataset", url)
+data = np.loadtxt(path)
+scaler = StandardScaler()
+data = scaler.fit_transform(data)
+
+"""
+## Affine coupling layer
+"""
+
+# Creating a custom layer with keras API
+
+
+nweights = 256
+reg = 0.01
+
+
+def Coupling(input_shape):
+    input = keras.layers.Input(shape=input_shape)
+
+    t_layer_1 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(input)
+    t_layer_2 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_1)
+    t_layer_3 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_2)
+    t_layer_4 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_3)
+    t_layer_5 = keras.layers.Dense(
+        2, activation="linear", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_4)
+
+    s_layer_1 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(input)
+    s_layer_2 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_1)
+    s_layer_3 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_2)
+    s_layer_4 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_3)
+    s_layer_5 = keras.layers.Dense(
+        2, activation="tanh", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_4)
+    return keras.Model(inputs=input, outputs=[s_layer_5, t_layer_5])
+
+
+Coupling(2).summary()
+"""
+## Real NVP
+"""
+# class containing the whole set of operations
```

Comment:
All comments throughout the file should start with a capital letter and end with a period. Please remove any comment whose content is redundant with what the code says